packages/cryptoplease/lib/features/activities/src/transaction_repository.dart

```diff
@@ -57,10 +61,11 @@ class TransactionRepository {
     );
     if (odp != null) return odp;
 
-    final oskp = await _db.oSKPRows.findActivityOrNull(
-      (row) => row.txId.equals(txId),
-      (pr) => pr.toActivity(_tokens),
-    );
+    final oskp = await _db.oSKPRows
```

Comment:
This is very expensive. You're fetching all the rows, deserializing them, and processing them on the Dart side.

Instead, we should store the tx ids for the statuses we need to track and match these ids with fetched txs.